undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDependencyProcessor.java

```diff
@@ -77,6 +86,15 @@ public void deploy(DeploymentPhaseContext phaseContext) {
         moduleSpecification.addSystemDependency(new ModuleDependency(moduleLoader, WEBSOCKET_API, false, false, true, false));
         moduleSpecification.addSystemDependency(new ModuleDependency(moduleLoader, JSTL, false, false, false, false));
 
+        final WarMetaData warMetaData = deploymentUnit.getAttachment(WarMetaData.ATTACHMENT_KEY);
+        if(warMetaData != null) {
+            final JBossWebMetaData metaData = warMetaData.getMergedJBossWebMetaData();
+            String servletContainerName = metaData.getServletContainerName();
+            if (servletContainerName == null) {
+                servletContainerName = defaultServletContainer;
+            }
+            phaseContext.addDeploymentDependency(UndertowService.SERVLET_CONTAINER.append(servletContainerName), UndertowAttachments.SERVLET_CONTAINER_SERVICE);
```

Comment:
SERVLET_CONTAINER_SERVICE is never put into attachment which causes test failures.